### DISCUSSION TITLE
What is the full list of parameters for worker resources?

### DISCUSSION THREAD

**Discussion by YarShev** [dask/dask#7058] [Q&A] [answered]
Hi there,

Having a look at the documentation page ([worker resources](https://distributed.dask.org/en/latest/resources.html)) I see only `GPU`, `MEMORY`, `process` parameters for worker resources. Can anyone show full list of parameters (if there are something else)?

Thanks in advance!

**Comment by mrocklin** [comment, marked as answer]
Resources are abstract, these are just suggestive names.  See
https://distributed.dask.org/en/latest/resources.html#resources-are-abstract

**Comment by YarShev** [comment]
Okay, I see. Thanks!
I have one more related question. Let's imagine the following code:
```python
class Foo():
    pass

from dask.distributed import Client
client = Client()

future = client.submit(Foo, actor=True)
```
Is it possible to specify a parameter that actor for `Foo` would occupy all resources of a node. For instance, `Foo` would place on one process but can use rest cores for threads?

**Reply by YarShev** [reply]
@mrocklin , are there any thoughts?

**Reply by YarShev** [reply]
@jsignell

**Reply by jsignell** [reply]
> You can use all other keywords to these functions like workers=, resources=, and so on to control where this actor ends up. [source](https://distributed.dask.org/en/latest/actors.html#create-an-actor)

If you have a worker per node with however many threads, then you should be able to accomplish what you are talking about.

**Reply by YarShev** [reply]
@jsignell , okay, thanks a lot!
Just to make sure. If I create an actor via `submit` as follows: actor = client.submit(Foo, actor=True, workers=<`concrete_worker_ip`>, resources={"threads": <`number_of_threads`>}), that will create actor on `concrete_worker_ip` with `number_of_threads` available. Am I correct?

**Reply by jsignell** [reply]
If you have set up the worker to have `nthreads` then yes. Otherwise if you have many workers and some of them have `nthreads`, then you can just set the `resources` on the `submit` and you will be guaranteed that your actor will go to a worker that has `nthreads`.
